test/e2e/e2e_test.go

```diff
@@ -358,6 +373,250 @@ var _ = ginkgo.Describe("Manager", ginkgo.Ordered, func() {
 			_, err = utils.Run(cmd)
 			gomega.Expect(err).NotTo(gomega.HaveOccurred())
 		})
+
+		ginkgo.It("deploy a working MCP server with auth policy enabled", func() {
+			mcpServerName := "everything"
+			var portForwardCmd *exec.Cmd
+			localPort := 8080
+
+			ginkgo.By("creating a secret with the JWKS")
+			// Read the JWKS content from the test data file
+			jwksContent, err := os.ReadFile("test/testdata/jwt/pub-key")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred(), "Failed to read JWKS file")
+
+			// Create the secret using kubectl with literal content (not base64 encoded)
+			cmd := exec.Command("kubectl", "create", "secret", "generic", "example-jwks",
+				"--from-literal=jwks="+string(jwksContent),
+				"-n", namespace)
+			_, err = utils.Run(cmd)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred(), "Failed to create JWKS secret")
+
+			ginkgo.By("creating an MCPServer for client testing")
+			mcpServer := &v1alpha1.MCPServer{
+				TypeMeta: v1.TypeMeta{
+					APIVersion: "kagent.dev/v1alpha1",
+					Kind:       "MCPServer",
+				},
+				ObjectMeta: v1.ObjectMeta{
+					Name:      mcpServerName,
+					Namespace: namespace,
+				},
+				Spec: v1alpha1.MCPServerSpec{
+					Deployment: v1alpha1.MCPServerDeployment{
+						Image: "docker.io/mcp/everything",
+						Port:  3000,
+						Cmd:   "npx",
+						Args:  []string{"-y", "@modelcontextprotocol/server-filesystem", "/"},
+					},
+					TransportType: "stdio",
+					Authentication: &v1alpha1.MCPServerAuthentication{
+						JWT: &v1alpha1.MCPServerJWTAuthentication{
+							Issuer:    "agentgateway.dev",
+							Audiences: []string{"test.agentgateway.dev"},
+							JWKS: &corev1.SecretKeySelector{
+								LocalObjectReference: corev1.LocalObjectReference{
+									Name: "example-jwks",
+								},
+								Key: "jwks",
+							},
+						},
+					},
+					Authorization: &v1alpha1.MCPServerAuthorization{
+						CEL: &v1alpha1.MCPServerCELAuthorization{
+							Rules: []string{
+								// Allow anyone to call 'read_file'
+								"mcp.tool.name == 'read_file'",
+								// Only the test-user can call 'write_file'
+								"jwt.sub == 'test-user' && mcp.tool.name == 'write_file'",
+								// Any authenticated user with the claim `nested.key == value` can access 'list_directory'
+								"mcp.tool.name == \"list_directory\" && jwt.nested.key == \"value\"",
```

Comment:
This test tests jwt authn + authz rules using the example jwks keys in the `test/testdata` directory. On line 508 onwards we parse the keys and initialize the client with the appropriate tokens to test access to the `list_directory` tool since only `example1.key` contains the claim `list_directory` should only by accessible when using `example1.key` jwt token.